k2/csrc/fsa_util.cc

```diff
@@ -11,34 +11,29 @@
 
 namespace k2 {
 
-void GetEnteringArcs(const Fsa &fsa, VecOfVec *entering_arcs) {
+void GetEnteringArcs(const Fsa &fsa, std::vector<int32_t> *arc_index,
+                     std::vector<int32_t> *end_index) {
   // CHECK(CheckProperties(fsa, KTopSorted));
 
   int32_t num_states = fsa.NumStates();
-  std::vector<std::vector<std::pair<Label, StateId>>> vec(num_states);
+  std::vector<std::vector<int32_t>> vec(num_states);
   int32_t num_arcs = 0;
+  int32_t k = 0;
   for (const auto &arc : fsa.arcs) {
-    auto src_state = arc.src_state;
     auto dest_state = arc.dest_state;
-    auto label = arc.label;
-    vec[dest_state].emplace_back(label, src_state);
+    vec[dest_state].push_back(k);
     ++num_arcs;
+    ++k;
```

Comment:
why not just use `num_arcs = fsa.arcs.size()` as `vector.size()` is always `O(1)`

Reply:
oh, yes, you are right. I will change it in the next commit.

Reply:
Fixed.